### ISSUE TITLE
Hide content in un-viewable forums

### ISSUE THREAD

**Issue by Sama34** [mybb/mybb#1106]
https://github.com/mybb/mybb/issues/679#issuecomment-48500546
- Frontend
  - [ ] Thread subscriptions should not be shown in the User CP.
  - [ ] Forum subscriptions should not be shown in the User CP.
  - [ ] Somehow saved drafts for inactive forums should have the forum hidden
  - [ ] Possibly allow the user to change the forum it will be posted in... (something that I think they should be able to do anyway)
  - [ ] User warning log in User CP
  - [ ] User post counts
  - [ ] Threads, posts should not be included in the statistics (including the number of posts the top posted has made, the averages, etc.) - ties into post counts
  - [x] Post reputation should verify forum permissions. #1239
- Mod CP
  - [ ] Reported posts within the forum should not be displayed in the header or Report Center.
  - [ ] Posts, threads and attachments should not be shown on the Mod CP Moderation Queue.
  - [ ] Moderator logs should not show for inactive forums
  - [ ] User warning log viewed by moderator
- Admin CP
  - [ ] Moderation Queue
  - [ ] Moderator Logs
  - [ ] Attachments
  - [ ] User Warning Log
  - [ ] Statistics include threads and post in inactive forums...

**Comment by JN-Jones** [comment]
Hiding information the user shouldn't see should be a priority and should be fixed ASAP @mybb/developers

**Comment by Stefan-ST** [comment]
However it doesn't reveal much. For example the user must have seen a thread at some point to subscribe it. So he already knows the subject (unless it has been changed later).
It would be high priority if it actually shows the posts or something like that.

**Comment by effone** [comment]
Are you still working on it @Sama34

**Comment by Eldenroot** [comment]
@effone I dont think so, @Sama34 had been away for a long time, now he is back but I am not sure if he will continue to work on all his assigned issues :( it would be nice :)

**Comment by Sama34** [comment]
@effone feel free to take any of my issues. Just send a message over discord or here for us to know.

**Comment by Ben-MyBB** [comment]
I don't think we should be looking at this stage to put this into 1.8. 1.9/1.10 @euantorano ?

**Comment by euantorano** [comment]
I'd be happy for smaller PRs fixing single points within this issue, but I wouldn't want to block any 1.8 release on this as such. It's kind of a hard one to assign a milestone to.

**Comment by Ben-MyBB** [comment]
@Sama34 ping

**Comment by Sama34** [comment]
Do we want this fixed for 1.8? I don't think this is high priority for 1.8, we can wait for 1.9 @mybb/developers

**Comment by euantorano** [comment]
I'm happy to wait until 1.9, but would also accept any small PRs to fix bits between now and then.

**Comment by Eldenroot** [comment]
Yeah, would be nice to solve at least some items from the list and get it into 1.8 :)

**Comment by Sama34** [comment]
Context: #4384

Soft-deleted content should not be shown in UserCP at all, even for admin or (super) moderators.

Ideally, soft-deleted content, being different to unapproved content, should only be displayed in moderation areas or areas where moderation actions can be executed upon them (i.e: search results, forum thread list, probably ModCP).

**Comment by Sama34** [comment]
Regarding the portal:

I think unapproved content should be displayed both in the latest threads side bar as well as announcement entries. This could be useful for administrators or (super) mods to identify content awaiting moderation.

Soft-deleted content shouldn't be displayed at all in portal since no moderation action can be executed upon them here.

**Comment by Sama34** [comment]
Also, the `get_visible_where()` function found in `inc/functions_search.php` should be moved to the main `inc/functions.php` file, as it will be useful to fix most of the points listed in this issue.
